app/code/Bulbulatory/Recommendations/Ui/Component/Listing/Column/Action.php

```diff
@@ -0,0 +1,52 @@
+<?php
+namespace Bulbulatory\Recommendations\Ui\Component\Listing\Column;
+
+use Magento\Framework\Escaper;
+use Magento\Framework\View\Element\UiComponent\ContextInterface;
+use Magento\Framework\View\Element\UiComponentFactory;
+use Magento\Ui\Component\Listing\Columns\Column;
+
+/**
+ * Class Action
+ */
+class Action extends Column
+{
+    protected $escaper;
+
+    protected $systemStore;
+    protected $customerLoader;
+
+    public function __construct(
+        ContextInterface $context,
+        UiComponentFactory $uiComponentFactory,
+        \Magento\Customer\Model\CustomerFactory $customerLoader,
+        Escaper $escaper,
+        array $components = [],
+        array $data = []
+    ) {
+        $this->escaper = $escaper;
+        $this->customerLoader = $customerLoader;
+        parent::__construct($context, $uiComponentFactory, $components, $data);
+    }
+
+    /**
+     * Prepare Data Source
+     *
+     * @param array $dataSource
+     * @return array
+     */
+    public function prepareDataSource(array $dataSource)
+    {
+        if (isset($dataSource['data']['items']))
+        {
+            foreach ($dataSource['data']['items'] as & $item)
+            {
+                $customer = $this->customerLoader->create()->load((int)$item[$this->getData('name')]);

```

Comment:
Magento Customer module offers a customer repository to easily retrieve customers - you can use it by including an object of type `CustomerRepositoryInterface` in the constructor. Then you can use it's `getById($customerId)` method. Also remember that not always you'll get a valid customer object - there may be a scenario when a customer sends a recommendation to someone and then his or her account gets deleted. You'll still have this customers ID in the database but it won't exist. `getById(..)` will throw a `NoSuchEntityException` which should be caught so the app won't break.

Reply:
@TM18

> Magento Customer module offers a customer repository to easily retrieve customers - you can use it by including an object of type CustomerRepositoryInterface in the constructor. Then you can use it's getById($customerId) method. Also remember that not always you'll get a valid customer object - there may be a scenario when a customer sends a recommendation to someone and then his or her account gets deleted. You'll still have this customers ID in the database but it won't exist. getById(..) will throw a NoSuchEntityException which should be caught so the app won't break.

Changed! Thanks for the new information!